File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int> ans={-1,-1};
        int first=search(nums,target,true);
        int last=search(nums,target,false);
        return {first,last};
    }
    int search(vector<int>& nums, int target, bool isleft){
        int left=0;
        int right= nums.size()-1;
        int ans=-1;
        while(left<=right){
            int mid = left + ((right - left)/2);
            if(nums[mid]>target) right=mid-1;
            else if(nums[mid]<target) left=mid+1;
            else{
                ans=mid;
                if(isleft){
                    right=mid-1;
                }
                else{
                    left= mid+1;
                }
            }
        }
        return ans;
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int search(int[] arr, int target , boolean isleft){
        int left = 0;
        int n = arr.length;
        int right = n - 1;
        int ans = -1;

        while(left <= right){
            int mid = left + ((right - left) / 2);
            if(arr[mid] < target) left = mid+1;
            else if(arr[mid] > target) right = mid - 1;
            else{
                ans = mid;
                if(isleft) right = mid -1;
                else left = mid + 1;
            }
        }
        return ans;
    }
    public int[] searchRange(int[] nums, int target) {
        int first = search(nums,target,true);
        int last = search(nums,target,false);
        return new int[] {first,last};
    }
}